packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -13,49 +13,23 @@ import {
 import { Icon, IIconProps } from '../../Icon';
 import { DirectionalHint } from '../../common/DirectionalHint';
 import { ContextualMenu, IContextualMenuProps } from '../../ContextualMenu';
-import { IButtonProps, IButton } from './Button.Props';
+import { IButtonProps, IButton, IButtonClassNames } from './Button.Props';
 import * as stylesImport from './BaseButton.scss';
 const styles: any = stylesImport;
 
-export interface IButtonClassNames {
-  base?: string;
-  variant?: string;
-  isDisabled?: string;
-  isEnabled?: string;
-  description?: string;
-  flexContainer?: string;
-  icon?: string;
-  menuIcon?: string;
-  label?: string;
-  root?: string;
-}
-
-/**
- * These props are not in the Props file as they are undocumented props only specific to BaseButton.
- *
- * @export
- * @interface IBaseButtonProps
- * @extends {IButtonProps}
- */
-export interface IBaseButtonProps extends IButtonProps {
-  /**
-   *  Custom class names for individual elements within the button DOM.
-   */
-  classNames?: IButtonClassNames;
-}
-
 export interface IBaseButtonState {
   menuProps?: IContextualMenuProps | null;
 }
 
-export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState> implements IButton {
+export class BaseButton extends BaseComponent<IButtonProps, IBaseButtonState> implements IButton {
 
-  public static defaultProps: IBaseButtonProps = {
+  public static defaultProps: IButtonProps = {
```

Comment:
So no exposing of the IButtonClassNames or any of that? Revert all the changes on the button files?